In [52]:
print("eda")

eda


In [53]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


In [54]:
df = pd.read_csv("../Data/SalesData.csv")
df.head()

,Order_ID,Date,Customer_ID,Product_Category,Product_Name,Quantity,Unit_Price_INR,Total_Sales_INR,Payment_Method,Delivery_Status,Review_Rating,Review_Text,State,Country
0,ORD100000,20-01-2025,CUST2796,Home & Kitchen,Cookware Set,2,25574.41,51148.82,Credit Card,Returned,1,Waste of money,Sikkim,India
1,ORD100001,28-08-2025,CUST9669,Beauty,Hair Dryer,1,19361.41,19361.41,Debit Card,Returned,5,Excellent product!,Telangana,India
2,ORD100002,27-02-2025,CUST5808,Electronics,Tablet,3,38476.22,115428.66,Cash on Delivery,Delivered,3,Fair deal,Nagaland,India
3,ORD100003,24-02-2025,CUST5889,Electronics,Headphones,5,38145.72,190728.60,Credit Card,Delivered,5,Highly recommend!,Assam,India
4,ORD100004,15-06-2025,CUST9005,Clothing,Saree,5,45940.98,229704.90,UPI,Delivered,5,Highly recommend!,Odisha,India


In [55]:
df['Payment_Method'].value_counts()

Payment_Method
Cash on Delivery    3827
Credit Card         3800
Debit Card          3727
UPI                 3646
Name: count, dtype: int64

In [68]:
df['State'].value_counts()

State
Sikkim               596
Rajasthan            568
Tamil Nadu           567
Meghalaya            559
Chhattisgarh         556
Manipur              555
Assam                553
Uttar Pradesh        551
Madhya Pradesh       550
Bihar                549
Jharkhand            547
Nagaland             545
Telangana            544
Odisha               542
Tripura              540
Karnataka            539
Maharashtra          531
West Bengal          528
Punjab               527
Goa                  522
Arunachal Pradesh    519
Haryana              518
Kerala               510
Mizoram              505
Gujarat              504
Uttarakhand          503
Himachal Pradesh     492
Andhra Pradesh       480
Name: count, dtype: int64

In [69]:
df.groupby("Review_Text")["Feedback"].value_counts().sort_values(ascending=False)

Review_Text                 Feedback
Satisfied with the product  Positive    658
Excellent product!          Positive    653
Fantastic quality!          Positive    641
Highly recommend!           Positive    637
Could be better             Neutral     632
Poor packaging              Negative    626
Worth every rupee!          Positive    615
Good quality                Positive    613
Average                     Neutral     611
Pretty decent               Positive    609
Quality could improve       Negative    607
Terrible experience         Negative    601
Not satisfied               Negative    600
Not bad                     Neutral     597
Would buy again             Positive    594
Okay product                Neutral     593
Product not as expected     Negative    592
Very bad product            Negative    586
Loved it!                   Positive    582
Fair deal                   Neutral     580
Disappointed                Negative    573
Waste of money              Negative   

In [56]:
for i in df.columns:
    print(i)
    print(df[i].nunique())

Order_ID
15000
Date
365
Customer_ID
7259
Product_Category
5
Product_Name
25
Quantity
5
Unit_Price_INR
14981
Total_Sales_INR
14988
Payment_Method
4
Delivery_Status
3
Review_Rating
5
Review_Text
25
State
28
Country
1


In [57]:
df['Product_Category'].value_counts()

Product_Category
Electronics       3036
Books             3035
Clothing          3022
Beauty            2997
Home & Kitchen    2910
Name: count, dtype: int64

In [58]:
df['Product_Name'].value_counts()

Product_Name
Children's Book     636
Perfume             629
Science Textbook    624
T-Shirt             623
Laptop              618
Tablet              616
Smartwatch          615
Air Fryer           614
Headphones          610
Jeans               604
Jacket              601
Hair Dryer          599
Fiction Novel       599
Saree               598
Comics              597
Sneakers            596
Face Cream          596
Lipstick            590
Shampoo             583
Mixer Grinder       582
Biography           579
Smartphone          577
Cookware Set        577
Water Purifier      570
Vacuum Cleaner      567
Name: count, dtype: int64

In [59]:
df.groupby("Product_Category")["Product_Name"].nunique()

Product_Category
Beauty            5
Books             5
Clothing          5
Electronics       5
Home & Kitchen    5
Name: Product_Name, dtype: int64

In [60]:
for category, group in df.groupby("Product_Category"):
    print(f"Category: {category}")
    print(group["Product_Name"].unique())

Category: Beauty
['Hair Dryer' 'Face Cream' 'Lipstick' 'Shampoo' 'Perfume']
Category: Books
['Comics' "Children's Book" 'Biography' 'Fiction Novel' 'Science Textbook']
Category: Clothing
['Saree' 'Jeans' 'Sneakers' 'T-Shirt' 'Jacket']
Category: Electronics
['Tablet' 'Headphones' 'Smartphone' 'Laptop' 'Smartwatch']
Category: Home & Kitchen
['Cookware Set' 'Vacuum Cleaner' 'Air Fryer' 'Mixer Grinder'
 'Water Purifier']


## Product Category Summary

The following observations were made from the product catalog:

- The platform offers **5 product categories**: Electronics, Clothing, Books, Beauty, and Home & Kitchen.
- Each product category contains **5 unique products**, resulting in a total of **25 products** across the platform.
- Customer purchases are **evenly distributed** across all products, with each product receiving approximately **570–636 orders**.
- No single product or category significantly outperformed the others, indicating a balanced customer purchasing pattern.

### Conclusion

The product catalog is well-structured, with an equal number of products in each category. Additionally, the nearly uniform purchase distribution suggests that customer demand is balanced across the entire product portfolio, making the dataset suitable for unbiased comparative analysis across different product categories.

In [61]:
df['Quantity'].value_counts()

Quantity
1    3114
5    2990
4    2985
2    2967
3    2944
Name: count, dtype: int64

In [62]:
df['Review_Rating'].value_counts()

Review_Rating
5    3128
4    3022
3    3013
2    2998
1    2839
Name: count, dtype: int64

In [63]:
df["Feedback"] = df["Review_Rating"].map({
    1: "Negative",
    2: "Negative",
    3: "Neutral",
    4: "Positive",
    5: "Positive"
})

In [64]:
df['Feedback'].value_counts()

Feedback
Positive    6150
Negative    5837
Neutral     3013
Name: count, dtype: int64

In [65]:
df['Review_Text'].value_counts()

Review_Text
Satisfied with the product    658
Excellent product!            653
Fantastic quality!            641
Highly recommend!             637
Could be better               632
Poor packaging                626
Worth every rupee!            615
Good quality                  613
Average                       611
Pretty decent                 609
Quality could improve         607
Terrible experience           601
Not satisfied                 600
Not bad                       597
Would buy again               594
Okay product                  593
Product not as expected       592
Very bad product              586
Loved it!                     582
Fair deal                     580
Waste of money                573
Disappointed                  573
Value for money               548
Defective item                543
Never buying again            536
Name: count, dtype: int64

In [66]:
for feedback, group in df.groupby("Feedback"):
    print(f"\n{'='*50}")
    print(f"{feedback}")
    print(f"{'='*50}")
    print(group["Review_Text"].unique())


Negative
['Waste of money' 'Poor packaging' 'Terrible experience'
 'Product not as expected' 'Never buying again' 'Very bad product'
 'Quality could improve' 'Disappointed' 'Defective item' 'Not satisfied']

Neutral
['Fair deal' 'Okay product' 'Average' 'Could be better' 'Not bad']

Positive
['Excellent product!' 'Highly recommend!' 'Fantastic quality!'
 'Worth every rupee!' 'Satisfied with the product' 'Good quality'
 'Would buy again' 'Loved it!' 'Pretty decent' 'Value for money']


## Customer Review & Feedback Summary

The following observations were made from the customer review data:

- Customers rated products on a **5-point review rating scale**, ranging from **1 (lowest)** to **5 (highest)**.
- The review ratings were categorized into **three feedback groups**: **Positive**, **Neutral**, and **Negative**.
- A total of **6,150 reviews** were classified as **Positive**, **3,013 reviews** as **Neutral**, and **5,837 reviews** as **Negative**.
- Each feedback category consists of meaningful customer review texts, allowing customer opinions to be analyzed more effectively.
- Positive feedback includes comments such as *"Excellent product!"*, *"Highly recommend!"*, and *"Value for money"*, while negative feedback highlights issues like *"Poor packaging"*, *"Waste of money"*, and *"Defective item"*.

### Conclusion

The customer reviews were successfully transformed into three meaningful feedback categories—**Positive**, **Neutral**, and **Negative**. This sentiment grouping simplifies customer opinion analysis and provides a strong foundation for identifying satisfaction trends, evaluating product performance, and generating actionable business insights.

In [67]:
df['Delivery_Status'].value_counts()

Delivery_Status
Delivered    5075
Pending      5044
Returned     4881
Name: count, dtype: int64